pkg/providers/vsphere/vsphere.go

```diff
@@ -1578,11 +1586,13 @@ func (p *vsphereProvider) ChangeDiff(currentSpec, newSpec *cluster.Spec) *types.
 }
 
 func (p *vsphereProvider) RunPostUpgrade(ctx context.Context, clusterSpec *cluster.Spec, managementCluster, workloadCluster *types.Cluster) error {
-	// Step 1: Refresh ClusterResourceSet
 	// This is unfortunate, but ClusterResourceSet's don't support any type of reapply of the resources they manage
 	// Even if we create a new ClusterResourceSet, if such resources already exist in the cluster, they won't be reapplied
-	// The long term solution is to add this capability to the cluster-api controller, with a new mode like "ReApplyOnChanges" or "ReApplyOnCreate" vs the current "ReApplyOnce"
-	// resourceSetManager.ForceUpdate(ctx, "clusterName-crs-0", constants.EksaSystemNamespace, managementCluster)
+	// The long term solution is to add this capability to the cluster-api controller,
```

Comment:
should we have a task for this?

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/5555